src/plone/restapi/services/multilingual/pam.py

```diff
@@ -92,17 +102,24 @@ def reply(self):
         self.request.response.setHeader("Location", self.context.absolute_url())
         return {}
 
-    def _traverse(self, url):
-        purl = getToolByName(self.context, "portal_url")
-        portal = purl.getPortalObject()
-        portal_url = portal.absolute_url()
-        if url.startswith(portal_url):
-            content_path = url[len(portal_url) + 1 :]
-            content_path = content_path.split("/")
-            content_item = portal.restrictedTraverse(content_path)
-            return content_item
-
-        return None
+    def get_object(self, key):
+        if isinstance(key, six.string_types):
```

Comment:
@erral may I ask why you check for a string here? What other possibilities do we have? What happens if this is not a string? Just wondering...

Reply:
I copied the code from the @copy endpoint: https://github.com/plone/plone.restapi/blob/master/src/plone/restapi/services/copymove/copymove.py#L27

Browsing in the history of changes, on previous versions we have an option to use intids as a way to identify objects:

https://github.com/plone/plone.restapi/blame/2c8fae7e412e6606f16617ab869f6b3c0b7d2c69/src/plone/restapi/services/copymove/copymove.py#L27

We removed the use of intids but didn't remove the string check:

https://github.com/plone/plone.restapi/commit/c8740e3e04f0fc66e8ae2362912235fd3beb290d

Which later was migrated to a six.string_types check.

I can do another PR remove that string_types check both here and in @copy endpoints.

Reply:
@erral that would be great. We can do it after the merge...